### PR TITLE
ASC-247 spacebindings inheritance

### PR DESCRIPTION
**Pull request by mfrancisc**
Covers acceptance criteria:
"The Space controller is updated so it inherits the SpaceBindings from the parent space and include them in the NSTemplateSet"
Jira story: https://issues.redhat.com/browse/ASC-247

Implement e2e space scenario to test:
- subSpace can inherit SpaceBindings from parentSpace
- changes to parentSpaceBindings are reflected to subSpace
- NSTemplateSet of a subSpace has username and roles from parentSpace + the specific ones from it's own SpaceBinding (if any)

Paired with:
- host-operator: https://github.com/codeready-toolchain/host-operator/pull/733

### CODE REVIEW

**Comment by mfrancisc** [comment]
/retest

Checking if happens again...
tried  the `parallel` test suite several times locally, but was unable to reproduce.

**Comment by mfrancisc** [comment]
/retest

no build logs or e2e test artifacts for this failure 😓

**Review by MatousJobanek — COMMENTED** [review]
Awesome :+1: 
it's like reading a story :star_struck: 
btw, the last use-case you could cover is creating a SpaceBinding in the sub-Space to override the SpaceBinding from the parent-Space

**Comment by mfrancisc** [comment]
> btw, the last use-case you could cover is creating a SpaceBinding in the sub-Space to override the SpaceBinding from the parent-Space
@MatousJobanek thanks for pointing that out, I've hopefully covered the case in  [7a4c695](https://github.com/codeready-toolchain/toolchain-e2e/pull/652/commits/7a4c695dbd36c74dee49206104efc0ec7a3e2572)

**Comment by mfrancisc** [comment]
/retest

infra, no build logs (again) 😓

**Comment by mfrancisc** [comment]
/retest

infra

File: test/e2e/parallel/space_test.go [hunk 1]
```diff
@@ -252,6 +252,167 @@ func TestRetargetSpace(t *testing.T) {
...
+					sort.Strings(sortedUsernames) // the NSTemplateSet has usernames sorted, this is required so that assertion doesn't fail when order changes
+					subSpaceNSTemplateSet, err = memberAwait.WaitForNSTmplSet(t, subSpaceNSTemplateSet.Name,
+						UntilNSTemplateSetHasConditions(Provisioned()),
+						UntilNSTemplateSetHasSpaceRoles(
+							SpaceRole(appstudioTier.Spec.SpaceRoles["admin"].TemplateRef, sortedUsernames...), // parent MUR is added as admin
+						),
+					)
+					require.NoError(t, err)
+					VerifyResourcesProvisionedForSpace(t, awaitilities, subSpace.Name)
+					// parentSpace should not have this username role added
```
Comment (MatousJobanek) [inline test/e2e/parallel/space_test.go:375]:
```suggestion
					// parentSpace should not be affected by the change in sub-space
```

File: test/e2e/parallel/space_test.go [hunk 2]
```diff
@@ -252,6 +252,167 @@ func TestRetargetSpace(t *testing.T) {
...
+						require.NoError(t, err)
+						subSpaceNSTemplateSet, err = memberAwait.WaitForNSTmplSet(t, subSpaceNSTemplateSet.Name,
+							UntilNSTemplateSetHasConditions(Provisioned()),
+							UntilNSTemplateSetHasSpaceRoles(
+								SpaceRole(appstudioTier.Spec.SpaceRoles["admin"].TemplateRef, parentSpaceBindings.Spec.MasterUserRecord), // removed admin role user anymore
+							),
+						)
+						require.NoError(t, err)
+						VerifyResourcesProvisionedForSpace(t, awaitilities, subSpace.Name)
+						// parentSpace should not have this username role added
```
Comment (MatousJobanek) [inline test/e2e/parallel/space_test.go:400]: // parentSpace should not be affected by the change in sub-space

**Review by MatousJobanek — APPROVED** [review]
Thanks for covering also the last use-case :+1:

**Pull request merged**